pkg/tsdb/elasticsearch/client/client.go

```diff
@@ -75,7 +83,7 @@ func coerceVersion(v *simplejson.Json) (*semver.Version, error) {
 }
 
 // NewClient creates a new elasticsearch client
-var NewClient = func(ctx context.Context, httpClientProvider httpclient.Provider, ds *models.DataSource, timeRange plugins.DataTimeRange) (Client, error) {
+var NewClient = func(ctx context.Context, httpClientProvider httpclient.Provider, ds *DatasourceInfo, timeRange backend.TimeRange) (Client, error) {
```

Comment:
Suggest parsing the data source configuration/JSONData in newInstanceSettings instead and put those as fields  on `DatasourceInfo`? Think code will be cleaner and good to stop using the simplejson package, [ref](https://github.com/grafana/grafana/blob/main/pkg/README.md#avoid-use-of-the-simplejson-package).

In hindsight and/or as a suggestion for the future. Think it could be a nice addition to refactor so that the whole client could be instantiated in the `newInstanceSettings` and by that making an instantiated client part of the data source instance. The only part restricting this right now is that the list of indices to be queried are embedded in the client. Don't remember right now if I had some other reason for making the client instantiated for each request rather than the lifecycle of a data source until being updated.